consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftBlockHeightManager.java

```diff
@@ -229,24 +256,30 @@ public void handleRoundChangePayload(final RoundChange message) {
       final RoundChangeArtifacts roundChangeMetadata = RoundChangeArtifacts.create(result.get());
 
       if (finalState.isLocalNodeProposerForRound(targetRound)) {
-        currentRound.startRoundWith(
-            roundChangeMetadata, TimeUnit.MILLISECONDS.toSeconds(clock.millis()));
+        if (currentRound.isEmpty()) {
+          startNewRound(0);
+        }
+        currentRound
+            .get()
+            .startRoundWith(roundChangeMetadata, TimeUnit.MILLISECONDS.toSeconds(clock.millis()));
       }
     }
   }
 
   private void startNewRound(final int roundNumber) {
```

Comment:
@jframe What do you think about addiing a check that roundNumber > currentRound?

Reply:
Don't think is necessary after all. The only way we can move to a arbitrary round is from a proposal or a round change. And both of these check the age of the message, so anything that for a prior round is logged and discarded by the handleRoundChangePayload on line 241 or by handleProposalPayload on line 186.